File: kurae-web/src/lib/types/referral-reward.ts
import type { DiscountType } from "@/lib/types/discount";

export type ReferralRewardSettings = {
  enabled: boolean;
  threshold: number;
  type: DiscountType;
  value: number;
};

export type BuyerReferralRewardItem = {
  code: string;
  rewardTier: number;
  type: DiscountType;
  value: number;
  grantedAt: string;
  redeemed: boolean;
};

export type BuyerReferralProgress = {
  sellerSlug: string;
  sellerName: string;
  code: string;
  successfulReferrals: number;
  threshold: number;
  rewardsEnabled: boolean;
  rewardType: DiscountType;
  rewardValue: number;
  progressInTier: number;
  referralsUntilReward: number;
  earnedRewards: BuyerReferralRewardItem[];
};

File: kurae-api/internal/domain/referral_reward.go
package domain

type ReferralRewardSettings struct {
	Enabled   bool         `json:"enabled"`
	Threshold int          `json:"threshold"`
	Type      DiscountType `json:"type"`
	Value     int          `json:"value"`
}

type BuyerReferralRewardItem struct {
	Code          string       `json:"code"`
	RewardTier    int          `json:"rewardTier"`
	Type          DiscountType `json:"type"`
	Value         int          `json:"value"`
	GrantedAt     string       `json:"grantedAt"`
	Redeemed      bool         `json:"redeemed"`
}

type BuyerReferralProgress struct {
	SellerSlug          string                    `json:"sellerSlug"`
	SellerName          string                    `json:"sellerName"`
	Code                string                    `json:"code"`
	SuccessfulReferrals int                       `json:"successfulReferrals"`
	Threshold           int                       `json:"threshold"`
	RewardsEnabled      bool                      `json:"rewardsEnabled"`
	RewardType          DiscountType              `json:"rewardType"`
	RewardValue         int                       `json:"rewardValue"`
	ProgressInTier      int                       `json:"progressInTier"`
	ReferralsUntilReward  int                       `json:"referralsUntilReward"`
	EarnedRewards       []BuyerReferralRewardItem `json:"earnedRewards"`
}
